lib/rebay_api_web/controllers/bid_controller.ex

```diff
@@ -6,9 +6,12 @@ defmodule RebayApiWeb.BidController do
 
   action_fallback RebayApiWeb.FallbackController
 
-  def index(conn, %{"item_uuid" => item_uuid}) do
+  plug RebayApiWeb.Plugs.AuthenticateSession when action in [:index_by_user, :create]
+  plug RebayApiWeb.Plugs.AuthorizeUser when action in [:index_by_user]
+
+  def index_by_item(conn, %{"item_uuid" => item_uuid}) do
```

Comment:
This may be pretty subjective.... but I don't think `index_*` is very descriptive. `index` has its own special meaning, and to me this is kinda like reading "root action for item", which imo doesn't really capture what this is doing. Let me know if you wanna discuss this further, but I figured I'd call it out again since this isn't really where I'd expect to find this behavior.

Reply:
Same comment goes for `index_by_user`.

Reply:
Right, I thought that might come up. I had been operating under a different concept of what `index` means--thinking of it as an action that pertains to listing resources. I thought it was common practice to add a dynamic param to an index path that provides filtering keys, where something like `/items` and `users/:id/items` both route to the same path, which filters the list depending on the presence of the `:id` key. That's what I had been asking if we should do rather than breaking the behavior up into two separate actions.

Reply:
But all that said, the requests that get routed to these actions still seemed to be the responsibility of an `index` action as I understood it. If i'm incorrect in that, then I'm happy renaming the actions, but if it's more subjective, then I'd want to push for either keeping it the way it is, or refactoring it as an action that is associated with multiple routes (`/items` `/users/:id/items`, etc.)

Reply:
> I thought it was common practice to add a dynamic param to an index path that provides filtering keys, where something like /items and users/:id/items both route to the same path, which filters the list depending on the presence of the :id key.

So in this case `/items` and `/users/:id/items` mean fairly different things, even though they both return "items". `/items` is the "index" of items, where you are acting directly on that resource. `/users/:id/items` on the other hand is acting on items that belong to a specific user. I don't know Phoenix convention around this, but I think the Rails convention would be something like:
```
/items => ItemsController#index
/users/:id/items` => UserItemsController#index (although I think DHH would name this like Users::ItemsController)
```

I'm personally a fan of splitting the actions/controllers because it allows you to do common validation at the controller level, and care less about how they may interact with all actions. Using the example above, `UserItemsControllers` can always check that the user is present and allowed to act on the requested resources, while `ItemsController` doesn't have to worry about it.

> ... which filters the list depending on the presence of the :id key.
I thought I'd just clarify here, path params should always be present. A query param should be used instead If the key/value is optional.

With all that said, I don't think this is a big deal, I just figured it should be called out as something to think about in the future.

Reply:
Okay that all makes sense, thanks! I think the missing piece in my understanding was the role of something like `UserItemsController`, and I can see how that would be better design.